### PR TITLE
Update priorities in font plugin

### LINKED ISSUE #35: Font feature should use lower priorities for their elements
The common conflict is underline + font color. Underline must be used inside font color because otherwise, the underline will not match the font color:

image

Right now it works in the editor just fine:

image

`<u>` is rendered inside `<span color>`. But this is just a coincident because `u` is sorted alphabetically after `span`. 

We should set font color's priority to a value between 6-9 (higher than link, lower than normal styles)... 7? :D 

I'm not sure about background color. I think it should have the same priority so both spans are nicely merged (`<span style=color:xxx; background-color: xxx>`). The remaining question is – do we change the priority for font-size and font-family?

I guess we must do that because of this:

image

### LINKED ISSUE #2291: Font feature should use lower priorities for their elements
The common conflict is underline + font color. Underline must be used inside font color because otherwise, the underline will not match the font color:

image

Right now it works in the editor just fine:

image

`<u>` is rendered inside `<span color>`. But this is just a coincident because `u` is sorted alphabetically after `span`. 

We should set font color's priority to a value between 6-9 (higher than link, lower than normal styles)... 7? :D 

I'm not sure about background color. I think it should have the same priority so both spans are nicely merged (`<span style=color:xxx; background-color: xxx>`). The remaining question is – do we change the priority for font-size and font-family?

I guess we must do that because of this:

image

### PR DESCRIPTION
**Pull request by msamsel**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Other: Update priorities in font plugin. Closes ckeditor/ckeditor5#2291.

---

### Additional information

* Priorities are set up to 7, to be higher than link (5), but lower than basic-styles (10). To have result that link remain most outer element and basic-styles elements will be always inside font features.

### CODE REVIEW

File: tests/utils.js [hunk 1]
```diff
@@ -97,14 +97,13 @@ describe( 'utils', () => {
...
-						},
-						priority: 5
+						}
 					}
 				},
 			] );
 		} );
 
-		it( 'adding colors table to dropdown works', () => {
+		it( 'addColorTableToDropdown()', () => {
```
Comment (jodator) [inline tests/utils.js]: I've just noticed it. The test are wrongly named... the `it()` part doesn't say much what the test does. It'd be nicer to structure the tests as:

```js
describe( 'utils', () => {
	describe( 'normalizeColorOptions()', () => {
		it( 'should return normalized config object' );
	} );
} );
```

or even

```js
describe( 'utils', () => {
	describe( 'normalizeColorOptions()', () => {
		it( 'should return normalized config object (string)' );

		it( 'should return normalized config object (object with name)' );

		it( 'should return normalized config object (object with name)' );
	} );
} );
```

The tests (as they test the same) might be put to util funciton and used to create thos test:

```js
testConfig( 'should return normalized config object (object)', { color: 'black' }, { /* output to assert */ } )
```

right now the tests doesn't say much what they test.

**Review by jodator — CHANGES_REQUESTED** [review]
The changes are OK I don't like the previous code for tests ;)

Small update but be better for the future when looking back on those tests.

**Comment by jodator** [comment]
@msamsel also could you check if you can do anything about those tests on edge? I hate that it renders attributes/styles in different order.

If there's and issue with writing one test for all browsers we probably sould create and issue for that (I also [had that issues](https://github.com/ckeditor/ckeditor5-mention/pull/76/commits/8f1c4c58b397d5a9ef5d0b37704cbe85f11ff60d)).

If you'd go with `env.isEdge` in tests to alter those checks then we're going to create an issue for this.

**Comment by msamsel** [comment]
> @msamsel also could you check if you can do anything about those tests on edge? I hate that it renders attributes/styles in different order.

I couldn't find nice solution for that, and I want to combine all 4 fonts' features to check if all are merged into single span. So `env.isEdge` seems to be only reasonable solution.

Alternatively I could use `match()`, however using complicated regular expression might significantly decrease readability in that test.

Should I create an issue for those sorting attributes problem?

**Comment by jodator** [comment]
> Should I create an issue for those sorting attributes problem?

Sure - it happens to me also so we should track it :)

**Comment by jodator** [comment]
@msamsel could you also do the `desribe() > it()` for other tests from that file? One `it()` fill be fine for each `it( 'someMethod()' )` which is still there.

**Review by jodator — APPROVED** [review]
LGTM :+1:
